ckanext/datagovuk/action/get.py

```diff
@@ -0,0 +1,65 @@
+import ckan.logic.schema
+from ckan.plugins.toolkit import (
+    abort,
+    check_access,
+    navl_validate,
+    ValidationError,
+    get_action,
+)
+import ckan.lib.dictization.model_save as model_save
+import ckan.lib.dictization.model_dictize as model_dictize
+
+from ckan.model import User
+
+
+log = __import__('logging').getLogger(__name__)
+
+
+# defined as they are in ckan/action/create.py to save further hacks to the
+# function copied from there
+_check_access = check_access
+_validate = navl_validate
+
+
+def user_auth(context, data_dict):
+    '''Authenticates a user
+
+    You must be a system administrator to authenticate users.
+
+    :param email: the email address of the user
```

Comment:
I can imagine some users might not remember which email address they logged in with.  I'm not sure usernames as such will be a thing in the future, I've not heard any mention of them so far.  Let's see what the 'Register an account' flow looks like.